spec/integration/twingly/amqp/subscription_spec.rb

```diff
@@ -275,6 +275,42 @@ def wait_for_messages_to_be_consumed
       end
     end
 
+    context "when a channel error occurs" do
+      def simulate_channel_error(channel)
+        channel_error_code = 406
+        channel_error_message =
+          "PRECONDITION_FAILED - delivery acknowledgement on channel #{channel.id} timed out."
+        channel.instance_variable_get(:@on_error).call(
+          channel,
+          AMQ::Protocol::Channel::Close.new(
+            channel_error_code,
+            channel_error_message,
+            0,
+            0
+          )
+        )
+      end
+
+      it "should call error callback" do
+        on_error_called = false
+        subject.on_error do |channel, channel_error|
+          on_error_called = true
+        end
+
+        simulate_channel_error(subject.raw_queue.channel)
+
+        expect(on_error_called).to eq(true)
+      end
```

Comment:
To make the test a bit shorter, and to also match the arguments sent to the callback, we can use a [`yield` matcher](https://rspec.info/features/3-13/rspec-expectations/built-in-matchers/yield/) :)

```suggestion
      it "should call error callback" do
        expect do |block|
          subject.on_error(&block)

          simulate_channel_error(subject.raw_queue.channel)

        end.to yield_with_args(subject.raw_queue.channel, be_kind_of(AMQ::Protocol::Channel::Close))
      end
```

Reply:
Neat 😎 Added in [da010ec](https://github.com/twingly/twingly-amqp/pull/106/commits/da010ecbaffaf691b963e678fa7560cec8b4e45b)